tests/network/network_service/test_service_cluster_ip.py

```
@@ -0,0 +1,46 @@
+        - ClusterIP Service exposing the server VM
+
+    Steps:
+        1. Start TCP server on the server VM
+        2. For each ClusterIP, establish TCP connection from client VM to server VM via the ClusterIP service
+
+    Expected:
+        - TCP connection through the ClusterIP service succeeds for all IP families
+    """
+    for service_ip in cluster_ip_service_for_server_vm.instance.spec.clusterIPs:
+        with subtests.test(msg=f"Testing connectivity via {service_ip}"):
+            with TcpServer(vm=service_server_vm, port=IPERF_SERVER_PORT) as server:
```

Ruff (0.15.18)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)